src/OpenTelemetry.Api/Trace/Tracer.cs

```diff
@@ -53,12 +55,15 @@ public static TelemetrySpan CurrentSpan
     }
 
     /// <summary>
-    /// Makes the given span as the current one.
+    /// Sets the given span as the current one in the context.
     /// </summary>
     /// <param name="span">The span to be made current.</param>
-    /// <returns>The current span.</returns>
+    /// <returns>The supplied span for call chaining.</returns>
+#if NET6_0_OR_GREATER
+    [return: NotNullIfNotNull(nameof(span))]
+#endif
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static TelemetrySpan WithSpan(TelemetrySpan span)
+    public static TelemetrySpan? WithSpan(TelemetrySpan? span)
```

Comment:
@alanwest This was another weird one. Doesn't seem to really be valid to pass a `null` in or get one back out but for all of these annotations I just tried to capture what was in place today and not change anything behaviors. Probably would have been better initially for this to throw a NRE for `null` input but it is what it is at this point 🤣